extensions/amp-fit-text/0.2/fit-text.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import * as styles from './fit-text.css';
+import {omit} from '../../../src/utils/object';
+import {px, setStyle} from '../../../src/style';
+import {useCallback, useEffect, useRef, useState} from '../../../src/preact';
+
+const {LINE_HEIGHT_EM_} = styles;
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function FitText(props) {
+  const {
+    'children': children,
+    'minFontSize': minFontSize = 6,
+    'maxFontSize': maxFontSize = 72,
+    'width': width = props['style']['width'] || '100%',
+    'height': height = props['style']['height'] || '100%',
+  } = props;
+  const rest = omit(props, [
```

Comment:
Do we allow rest props yet? This is equivalent to:

```js
const {
    'children': …,
    'minFontSize': …,
    'maxFontSize': …,
    'width': width = …,
    'height': height = …,
    ...rest
} = props;

Reply:
I get this [eslint error](https://github.com/ampproject/amphtml/blob/master/.eslintrc#L118): `Collecting props using a rest property is not allowed - local/no-rest`

Not sure what you mean by "do we allow it" -- are you suggesting to remove the rule or is there more work to be done here?

Reply:
I'd really like to have `...rest`. @jridgewell can we enable?

Reply:
Yes, I think we can remove the eslint restriction for props. We cannot allow it for arrays, though, due to the iteration polyfill it requires.

Reply:
What's the best way to modify the rules to only allow for props?

Reply:
Just need to delete https://github.com/ampproject/amphtml/blob/34c3f78535608d7cdd3061756088308561891764/build-system/eslint-rules/no-rest.js#L40-L45